step3.nim

```diff
@@ -88,11 +88,12 @@ proc mainLoop() {.cdecl.} =
   glClear(GL_COLOR_BUFFER_BIT)
 
   m = rotateZ(getTime().float32)
-  p = ortho(-1, 1, 1, -1, -1000, 1000)
+  p = ortho[float32](-1, 1, 1, -1, -1000, 1000)
   mvp = m * p;
 
   glUseProgram(program)
-  glUniformMatrix4fv(mvpLocation.GLint, 1, GL_FALSE, addr mvp[0]);
+  var mvp_addr = cast[ptr GLFloat](addr mvp[0])
```

Comment:
if you don't use `--define:vmathArrayBased` does `cast[ptr GLFloat](mvp.unsafeAddr)` work?

Reply:
That did work actually! I made the change to unsafeAddr and removed the --define from `step3.nims`.